docs/source/development/commands_reference.md

```diff
@@ -492,6 +493,14 @@ The command also accepts an optional `--pipeline` argument that allows you to sp
 kedro catalog list --pipeline=ds,de
 ```
 
+##### Resolve dataset factories in the catalog
+
+```bash
+kedro catalog resolve
+```
+
+This command resolves dataset factories in the catalog file with any explicit entries in the pipeline. The output is the aggregation of all explicit dataset entries in the catalog and factory datasets in the pipeline.
```

Comment:
I would say something like "This command shows a fully resolved catalog. It includes datasets explicitly mentioned in your catalog files and datasets mentioned in the project's pipelines resolved against corresponding dataset factory patterns". (Not exactly this but something to this effect)